whois-db/src/main/java/net/ripe/db/whois/db/WhoisServer.java

```diff
@@ -84,6 +88,12 @@ public void start() {
     }
 
     public void stop() {
+        try {
+            Thread.sleep(preShutdownPause * 1000);
```

Comment:
Is this necessary? Will the WhoisServer not wait for Jetty to shutdown first anyway?

Reply:
Yes it turns out to be necessary, and no unfortunately the WhoisServer does not wait for Jetty to shutdown first. Not sure if there is another way to enforce this

Reply:
OK, can you add a TODO explaining why this sleep is there. Also I suggest adding an extra delay here, so there isn't a race condition with the DelayShutdownHook (if the delay is the same, the WhoisServer may exit first?). Also log here if the wait is interrupted.

Reply:
I added some explanatory texts. I do not think there is a need to add an extra delay. There won't be a race condition; worst case scenario is the health check endpoint stays available for couple of milliseconds/seconds shy of the actual delay.

Reply:
OK! Thanks